gui/src/components/Sidebar.tsx

```diff
@@ -0,0 +1,216 @@
+    name: "Contracts",
+    component: Contracts,
+    devOnly: true,
+    icon: CallToActionIcon,
+  },
+  {
+    path: "connections",
+    name: "Connections",
+    component: Peers,
+    icon: OnlinePredictionSharpIcon,
+  },
+];
+
+const WIDTH_MD = 200;
+const WIDTH_SM = 80;
+
+export function Sidebar({ children }: { children: ReactNode }) {
+  const [_match, params] = useRoute("/:path");
+  const [_location, setLocation] = useLocation();
+  const { theme } = useTheme();
+
+  const handleKeyboardNavigation = (event: KeyboardEvent) => {
+    setLocation(TABS[parseInt(event.key) - 1].path);
+  };
+
+  useMenuAction((payload) => setLocation(payload));
+
+  useKeyPress(
+    range(1, TABS.length + 1).map(toString),
+    { meta: true },
+    handleKeyboardNavigation
+  );
+
+  useKeyPress(
+    range(1, TABS.length + 1).map(toString),
+    { ctrl: true },
+    handleKeyboardNavigation
+  );
+
+  return (
+    <Box>
+      <Drawer
+        PaperProps={{
+          variant: "lighter",
+          sx: {
+            width: WIDTH_MD,
+            [theme.breakpoints.down("md")]: {
+              width: WIDTH_SM,
+              justifyContent: "center",
+            },
+          },
+        }}
+        sx={{ flexShrink: 0 }}
+        variant="permanent"
+      >
+        <Box
+          flexGrow={1}
+          display="flex"
+          flexDirection="column"
+          sx={{
+            [theme.breakpoints.down("md")]: {
+              alignItems: "center",
+            },
+          }}
+        >
+          <Box flexShrink={0} sx={{ px: 2, pt: 6 }}>
+            <Logo width={40} />
+          </Box>
+          <Stack p={2} rowGap={1} flexGrow={1}>
+            {TABS.map((tab, index) => (
+              <SidebarTab
+                key={index}
+                tab={tab}
+                selected={
+                  index === Math.max(findIndex(TABS, { path: params?.path }), 0)
+                }
+              />
+            ))}
+          </Stack>
+          <Stack
+            rowGap={1}
+            p={2}
+            sx={{
+              [theme.breakpoints.down("md")]: {
+                display: "none",
+              },
+            }}
+          >
+            <QuickWalletSelect />
+            <QuickAddressSelect />
+            <QuickNetworkSelect />
+          </Stack>
+          <Stack
+            p={2}
+            rowGap={1}
+            sx={{
+              [theme.breakpoints.down("md")]: {
+                justifyContent: "center",
+              },
+            }}
+          >
+            <CommandBarButton />
+            <SettingsButton />
+          </Stack>
+        </Box>
+      </Drawer>
+      <Box
+        sx={{
+          pl: `${WIDTH_MD}px`,
+          [theme.breakpoints.down("md")]: {
+            pl: `${WIDTH_SM}px`,
+          },
+        }}
+      >
+        {children}
+      </Box>
+    </Box>
+  );
+}
+
+function SidebarTab({
+  tab,
+  selected,
+}: {
+  tab: (typeof TABS)[number];
+  selected: boolean;
+}) {
```

Comment:
can you move this to a `SidebarTabProps` interface?

Reply:
I can, but do you mind me asking why? I'm not against it; just trying to understand your reasoning. I usually only export these to interfaces when they become messy.

Reply:
I don't mean exporting it, just declaring it separately from the component.
I find this syntax honestly terrible to read. it's fine'ish when it's just one element (such as `{ children: React.Node }`), but anything more becomes really ugly

Reply:
👍